pkg/trustedresources/verify.go

```diff
@@ -168,11 +238,37 @@ func getVerifiers(ctx context.Context, k8s kubernetes.Interface) ([]signature.Ve
 			verifiers = append(verifiers, v...)
 		}
 	}
+
+	return verifiers
+}
+
+// verifiersFromPolicy get all verifiers from VerificationPolicy.
+// For each policy, loop the Authorities of the VerificationPolicy to fetch public key
+// from either inline Data or from a SecretRef.
+func verifiersFromPolicy(ctx context.Context, k8s kubernetes.Interface, policy *v1alpha1.VerificationPolicy) ([]signature.Verifier, error) {
+	verifiers := []signature.Verifier{}
+	for _, a := range policy.Spec.Authorities {
+		algorithm, err := hashAlgorithm(a.Key.HashAlgorithm)
+		if err != nil {
+			return nil, fmt.Errorf("authority %q contains an invalid hash algorithm: %w", a.Name, err)
+		}
+		if a.Key.Data != "" {
+			verifier, err := verifierFromKeyData([]byte(a.Key.Data), algorithm)
+			if err == nil {
```

Comment:
similar comment below, seems like we're ignoring the error here? i.e. if there is an error i think we should stop and return an error?